### PR TITLE
New `admin check` command structure implementation

### PR DESCRIPTION
**Pull request by kevinrr888**
This is part of https://github.com/apache/accumulo/issues/4687
Changes:
- Added new `admin check` command
- `admin check list` prints the checks that can be run along with their descriptions and dependencies
- `admin check run` runs all the checks or a specified list of checks (explicitly provide each check, provide a regex of the checks to run, or just run all if no further args are provided). The dependencies are run first and if a checks dependency fails, the check is skipped
- Current impl of `admin check run` does not do any actual work besides printing that the check is running (to ensure correct run order) and returning an `OK` status
- New IT AdminCheckIT
	- This IT includes a verion of the Checks where no work is actually done (right now, this is equivalent to the actual implementation). This is done so correct run order, correct checks run, etc. can be
	verified without actually running the checks which may take a long time. More tests can be added later to test the actual check functionality when that is implemented.
- No existing functionality was changed

Questions:
1) Are new classes and packages in suitable locations?
2) Is the way I went about implementing the dummy checks in AdminCheckIT okay? Mostly concerned if changing the real static CHECK_RUNNERS field is okay (changes it during the test but asserts that it is reset after the test completes).

### CODE REVIEW

**Comment by kevinrr888** [comment]
Some example outputs and a sketch of the current dependency tree as support:
```
Dependency tree sketch:

        SYSTEM_CONFIG
              |
              V
        ROOT_METADATA
              |
              V
          ROOT_TABLE
         /          \
   METADATA_TABLE  SYSTEM_FILES
         |
         V
     USER_FILES
```
Sample outputs:

```
$ accumulo admin check list

Check Name           | Description                                                                      | Depends on          
------------------------------------------------------------------------------------------------------------------------
SYSTEM_CONFIG        | Validate the system config stored in ZooKeeper                                   |                     
ROOT_METADATA        | Checks integrity of the root tablet metadata stored in ZooKeeper                 | SYSTEM_CONFIG       
ROOT_TABLE           | Scans all the tablet metadata stored in the root table and checks integrity      | ROOT_METADATA       
METADATA_TABLE       | Scans all the tablet metadata stored in the metadata table and checks integrity  | ROOT_TABLE          
SYSTEM_FILES         | Checks that files in system tablet metadata exist in DFS                         | ROOT_TABLE          
USER_FILES           | Checks that files in user tablet metadata exist in DFS                           | METADATA_TABLE      
------------------------------------------------------------------------------------------------------------------------
```

```
$ accumulo admin check run

Running check SYSTEM_CONFIG
Check SYSTEM_CONFIG completed with status OK
Running check ROOT_METADATA
Check ROOT_METADATA completed with status OK
Running check ROOT_TABLE
Check ROOT_TABLE completed with status OK
Running check METADATA_TABLE
Check METADATA_TABLE completed with status OK
Running check SYSTEM_FILES
Check SYSTEM_FILES completed with status OK
Running check USER_FILES
Check USER_FILES completed with status OK

Check Name           | Status              
--------------------------------------------------
SYSTEM_CONFIG        | OK                  
ROOT_METADATA        | OK                  
ROOT_TABLE           | OK                  
METADATA_TABLE       | OK                  
SYSTEM_FILES         | OK                  
USER_FILES           | OK                  
--------------------------------------------------
```

```
$ accumulo admin check run -p ".*table"

Running check SYSTEM_CONFIG
Check SYSTEM_CONFIG completed with status OK
Running check ROOT_METADATA
Check ROOT_METADATA completed with status OK
Running check ROOT_TABLE
Check ROOT_TABLE completed with status OK
Running check METADATA_TABLE
Check METADATA_TABLE completed with status OK

Check Name           | Status              
--------------------------------------------------
SYSTEM_CONFIG        | OK                  
ROOT_METADATA        | OK                  
ROOT_TABLE           | OK                  
METADATA_TABLE       | OK                  
--------------------------------------------------
```

In this example, I made ROOT_TABLE fail when it was run and attempted to run all the checks
```
$ accumulo admin check run

Running check SYSTEM_CONFIG
Check SYSTEM_CONFIG completed with status OK
Running check ROOT_METADATA
Check ROOT_METADATA completed with status OK
Running check ROOT_TABLE
Check ROOT_TABLE completed with status FAILED

Check Name           | Status              
--------------------------------------------------
SYSTEM_CONFIG        | OK                  
ROOT_METADATA        | OK                  
ROOT_TABLE           | FAILED              
METADATA_TABLE       | SKIPPED_DEPENDENCY_FAILED
SYSTEM_FILES         | SKIPPED_DEPENDENCY_FAILED
USER_FILES           | SKIPPED_DEPENDENCY_FAILED
--------------------------------------------------
```

File: server/base/src/main/java/org/apache/accumulo/server/util/Admin.java [hunk 1]
```diff
@@ -1113,4 +1206,129 @@ private static void getAllFateIds(TabletMetadata tabletMetadata,
...
+    } else {
+      // run without regex provided
+      if (cmd.checks == null) {
+        cmd.checks = EnumSet.allOf(CheckCommand.Check.class).stream().map(Enum::name)
+            .collect(Collectors.toList());
+      }
+    }
+  }
+
+  private static void listChecks() {
```
Comment (ddanielr) [inline server/base/src/main/java/org/apache/accumulo/server/util/Admin.java:1254]: Is there a reason you didn't want to use a logger here?

File: server/base/src/main/java/org/apache/accumulo/server/util/Admin.java [hunk 2]
```diff
@@ -1113,4 +1206,129 @@ private static void getAllFateIds(TabletMetadata tabletMetadata,
...
+      if (!allChecksToRun.contains(check)) {
+        allChecksToRun.add(check);
+        computeAllChecksToRun(check.getDependencies(), allChecksToRun);
+      }
+    }
+  }
+
+  private static void
+      printChecksResults(TreeMap<CheckCommand.Check,CheckCommand.CheckStatus> checkStatus) {
+    System.out.println();
```
Comment (ddanielr) [inline server/base/src/main/java/org/apache/accumulo/server/util/Admin.java:1325]: Same logger question here

File: server/base/src/main/java/org/apache/accumulo/server/util/Admin.java (hunk 1 above)
Comment (kevinrr888) [reply server/base/src/main/java/org/apache/accumulo/server/util/Admin.java:1254]: If we want the output to go to a file, we can just redirect the output there.

**Comment by kevinrr888** [comment]
Closing this to be replaced by https://github.com/apache/accumulo/pull/4807.
As suggested by @ctubbsii, making these changes target main

**Pull request closed without merging**